### PR TITLE
Fixes #33361 - Initial stab at the host errata page

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by parthaa** [comment]
[test katello]

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import {
+  BugIcon,
+  SecurityIcon,
+  EnhancementIcon,
+} from '@patternfly/react-icons';
+
+import IsoDate from 'foremanReact/components/common/dates/IsoDate';
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import PropTypes from 'prop-types';
+import TableWrapper from '../../../../components/Table/TableWrapper';
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:16]: The top bar of TableWrapper (the search and pagination) needs margin on the top and bottom to look more like the mockup. Right now it's very crowded. I'm thinking the solution should be in a `TableWrapper.scss` so we don't have to keep duplicating the CSS in every table..

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            return (
+              <Tr key={`${id}_${createdAt}`}>
+                <Td>
+                  <a href={urlBuilder(`errata/${id}`, '')}>{errataId}</a>
+                </Td>
+                <Td><ErrataType {...erratum} /></Td>
+                <Td><ErrataSeverity {...erratum} /></Td>
+                <Td><TableText wrapModifier="truncate">{title}</TableText></Td>
+                <Td key={publishedAt}><IsoDate date={publishedAt} /></Td>
+              </Tr>
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:151]: Should add the action menu and (disabled) action items here, at the end of each table row

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          metadata,
+          emptyContentTitle,
+          emptyContentBody,
+          emptySearchTitle,
+          emptySearchBody,
+          searchQuery,
+          updateSearchQuery,
+          status,
+          }}
+      fetchItems={fetchItems}
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:122]: Should add the disabled "Apply" button here, via an `actionButtons` prop.  Also add the action menu and disabled action items.

**Review by jeremylenz — COMMENTED** [review]
@parthaa Awesome work!  New TableWrapper is working great :smile: 

Initial comments below

File: webpack/components/extensions/HostDetails/HostErratum/HostErratumActions.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { API_OPERATIONS, get } from 'foremanReact/redux/API';
+import { foremanApi } from '../../../../services/api';
+import { HOST_ERRATUM_KEY } from './HostErratumConstants';
```
[the remark this answers is not in the document]
Comment (parthaa) [reply webpack/components/extensions/HostDetails/HostErratum/HostErratumActions.js]: updated

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      label = __('Important');
+      break;
+    case 'Critical':
+      color = 'red';
+      label = __('Critical');
+      break;
+    default:
+      label = __('N/A');
+  }
+  return <TableText wrapModifier="nowrap">{color && <SecurityIcon color={color} title={label} />} {label} </TableText>;
```
[the remark this answers is not in the document]
Comment (parthaa) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:74]: its basically saying dont show a securityIcon if no color was set.  The label will be `N/A` in that case.

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js (hunk 1 above)
Comment (Andrewgdewar) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:16]: If we do this, that scss likely needs to be conditional. 
I have pages that use the table and expect little to no margin for example.
https://marvelapp.com/prototype/1121a8de/screen/72413287

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+import PropTypes from 'prop-types';
+import TableWrapper from '../../../../components/Table/TableWrapper';
+import { getInstallableErrata } from '../HostErrata/HostErrataActions';
+import { selectHostErrataStatus } from '../HostErrata/HostErrataSelectors';
+import { HOST_ERRATUM_KEY } from '../HostErrata/HostErrataConstants';
+
+export const ErrataType = ({ type }) => {
+  let ErrataIcon;
+  let label;
+  switch (type) {
```
Comment (Andrewgdewar) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:24]: Good use of switch

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js (hunk 3 above)
Comment (parthaa) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:55]: updated

**Comment by parthaa** [comment]
@Andrewgdewar @jeremylenz CAn you give a second look. I think I ve addressed all of the comments.

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js (hunk 2 above)
Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:87]: @parthaa thoughts on adding this?

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab.js (hunk 1 above)
Comment (jeremylenz) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab.js:10]: @parthaa I'm still seeing no margin between the tab bar and the TableWrapper top bar.  I agree with @Andrewgdewar that the styles should probably be per table.

**Review by jeremylenz — COMMENTED** [review]
@parthaa Changes look good, but looks like a couple are still pending? (See my replies above)

**Comment by parthaa** [comment]
errata--paging

File: webpack/components/extensions/HostDetails/Tabs/__tests__/errataTab.test.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  assertNockRequest(searchDelayScope);
+  assertNockRequest(autoSearchScope);
+});
+
+test('Can call API for errata and show on screen on page load', async (done) => {
+  // Setup autocomplete with mockForemanAutoComplete since we aren't adding /katello
+  const autocompleteScope = mockForemanAutocomplete(nockInstance, autocompleteUrl);
+
+  // Mocking our host ID, results, and status in the test redux store
+  const useSelector = jest.spyOn(reactRedux, 'useSelector');
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/__tests__/errataTab.test.js:40]: I found that mocking useSelector caused some super-weird problems.  Instead I recommend adding to the renderOptions like I ended up doing here: https://github.com/Katello/katello/pull/9594/files#diff-0aced2c1e6b8cecf1ff1e000ed939f3ca5ac19949a73e0188301cdb3f710a02f

**Review by jeremylenz — COMMENTED** [review]
Per our discussion at today's meeting, let's add a banner with a link to the old Errata page :)

**Comment by parthaa** [comment]
> Per our discussion at today's meeting, let's add a banner with a link to the old Errata page :)

But I thought we were not going to do that for Errata ?

**Comment by jeremylenz** [comment]
> But I thought we were not going to do that for Errata ?

Right, we're going to finish it instead :)

I was just thinking put the banner for now, and then remove it when we have full functionality of the tab?  Since that's going to be the standard for all the tabs I figured it might be good to start this way.

**Comment by parthaa** [comment]
How does this look?

errata--paging

**Comment by parthaa** [comment]
Update to
errata--paging

**Review by jeremylenz — APPROVED** [review]
Changes look great, just fix eslint & tests and we're good to go :)

APJ :+1:

**Comment by jeremylenz** [comment]
@parthaa The link is styled as a `<button>` element so it doesn't do anything when you click it.  I rescind my ACK..

**Review by jeremylenz — CHANGES_REQUESTED** [review]
fix button link thx!

**Comment by parthaa** [comment]
> fix button link thx!

updated

**Review by jeremylenz — APPROVED** [review]
ACK :+1: 

Nice work @parthaa!

**Pull request merged**